src/cli.rs

```diff
@@ -210,6 +213,24 @@ impl Rover {
         }
     }
 
+    fn check_markdown_available(&self, command: &Command) -> Result<()> {
+        if self.output_type != OutputType::Markdown {
+            return Ok(());
+        }
+
+        match command {
+            Command::Subgraph(sub_cmd) => match sub_cmd.command {
+                command::subgraph::Command::Check(_) => Ok(()),
+                _ => Err(RoverError::new(anyhow!(
+                    "Your command is not available with markdown output."
+                ))),
+            },
+            _ => Err(RoverError::new(anyhow!(
+                "Your command is not available with markdown output."
+            ))),
+        }
```

Comment:
if there is a better way to traverse sub command, please let me know.

Reply:
maybe the `matches!` macro could be a bit cleaner? (i haven't tested if this code works or not but this is about what i'd expect)

```rust
let is_check = match command {
  Command::Subgraph(subcommand) => matches!(subcommand, command::subgraph::Command::Check(_)),
  Command::Graph(subcommand) => matches!(subcommand, command::graph::Command::Check(_)),
  _ => false
};
if is_check {
  Ok(())
} else {
  Err(RoverError::new(anyhow!("This command does not support markdown output.")))
}
```